### PR TITLE
Setup Test bundle for service catalog tests and do a clean up after them

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Setup Test bundle for service catalog tests and do a clean up after them

**Related issue(s)**
See #472

### CODE REVIEW

File: tests/ui-tests/tests/catalog-ui-test.js [hunk 1]
```diff
@@ -8,12 +8,24 @@ import address from '../utils/address';
 import { describeIf } from '../utils/skip';
 import dex from '../utils/dex';
 
+import { TestBundleInstaller } from '../setup/test-bundle-installer';
+
+const TEST_NAMESPACE = 'service-catalog-ui-test';
```
Comment (magicmatatjahu) [inline tests/ui-tests/tests/catalog-ui-test.js]: Pleas extract this const to Config

File: tests/ui-tests/setup/namespace-manager.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+export class NamespaceManager {
+  constructor(namespace) {
+    const kubeConfig = loadKubeConfig();
+    this.api = kubeConfig.makeApiClient(k8s.Core_v1Api);
+    this.namespaceName = namespace;
+  }
+
+  async createIfDoesntExist() {
+    if (await this.exists) {
```
Comment (magicmatatjahu) [inline tests/ui-tests/setup/namespace-manager.js]: `await this.exists()`?

File: tests/ui-tests/setup/namespace-manager.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      console.log(
+        `Namespace ${this.namespaceName} not found. Skipping deleting it...`
+      );
+    }
+
+    console.log(`Deleting namespace ${this.namespaceName}...`);
+    await this.api.deleteNamespace(this.namespaceName);
+  }
+
+  getNamespaceObj() {
```
Comment (magicmatatjahu) [inline tests/ui-tests/setup/namespace-manager.js]:
``` js
get namespaceObj() {
   return {
      metadata: {
        name: this.namespaceName
      }
    };
}
```

File: tests/ui-tests/setup/namespace-manager.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      );
+      return;
+    }
+
+    console.log(`Creating namespace ${this.namespaceName}...`);
+    await this.api.createNamespace(this.getNamespaceObj());
+  }
+
+  async deleteIfExists() {
+    if (!(await this.exists)) {
```
Comment (magicmatatjahu) [inline tests/ui-tests/setup/namespace-manager.js]: `await this.exists()`?

File: tests/ui-tests/tests/catalog-ui-test.js (hunk 1 above)
Comment (pkosiec) [reply tests/ui-tests/tests/catalog-ui-test.js]: I did it intentionally. I think we shouldn't put everything in config. This is a value needed only for the `catalog-ui-test`, so it makes sense for me to define it in the particular file.

File: tests/ui-tests/setup/test-bundle-installer.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import { NamespaceManager } from './namespace-manager';
+import { HelmBrokerConfigurer } from './helm-broker/configurer';
+
+export class TestBundleInstaller {
+  constructor(namespace) {
+    this.kubeConfig = loadKubeConfig();
+    this.namespaceName = namespace;
+    this.namespaceManager = new NamespaceManager(namespace);
+    this.helmBrokerConfigurer = new HelmBrokerConfigurer();
+  }
```
[the remark this answers is not in the document]
Comment (pkosiec) [reply tests/ui-tests/setup/test-bundle-installer.js]: Good catch. I wanted to do so after my modifications, but I forgot it. Visual Studio Code doesn't handle autocompletion for JS files, when doing DI with constructor 😞

File: tests/ui-tests/setup/namespace-manager.js (hunk 3 above)
Comment (pkosiec) [reply tests/ui-tests/setup/namespace-manager.js]: I will keep it as it is, as vscode doesn't handle getters autocompletion in pure JS. It's definitely safer. When we move to TypeScript - no problem, I will refactor it.

Comment (magicmatatjahu) [reply tests/ui-tests/setup/namespace-manager.js]: Ok 😢

File: tests/ui-tests/tests/catalog-ui-test.js (hunk 1 above)
Comment (magicmatatjahu) [reply tests/ui-tests/tests/catalog-ui-test.js]: Let it be...

Comment (pkosiec) [reply tests/ui-tests/tests/catalog-ui-test.js]: https://www.youtube.com/watch?v=2xDzVZcqtYI

**Pull request merged**